docs/config/README.md

```diff
@@ -2236,6 +2247,9 @@ case the module will only be active in directories that match those conditions.
 [kubernetes]
 format = 'on [⛵ ($user on )($cluster in )$context \($namespace\)](dimmed green) '
 disabled = false
+environments = [
```

Comment:
Alternative would be "contexts" (and renaming `context_pattern` to `pattern`)

Reply:
@jankatins I don't have lots of experience with Kubernetes, but some searching suggests that "contexts" is the less ambiguous (and more idiomatic) term for Kubernetes. Does that square with your experience? If so, I would vote for going with contexts and rename the pattern variable. The term "environment" is overloaded enough as it is.

Would also be interested in @nomaed's opinion on this.

As far as the aliases go, I think that sounds like a good idea (but again, I don't know much about Kubernetes). It seems to me that it basically eliminates the need for multiple sets regexes, which are a little fiddly to write. In the worst case, we can deprecate the user_alias and context_alias options now, then start emitting warnings for anyone using them in a future version, before finally getting rid of them for 2.0.

Reply:
#4505 also seems to suggest that "contexts" is the preferred term.

Reply:
So:
-Rename environment to context
- Add user_pattern, user_alias, context_alias (given that we have user and context pattern, I would keep the `context_` prefix to `context_pattern` -> user_pattern would only be evaluated if set.
- Maker that the default way to set user and context aliases: if not set via the matched context
- look at the old way and omit a warning
- If can, also port the "set alias from regex group" feature.

Reply:
Done now, apart from the regex group feature